operator/controllers/btpoperator_controller.go

```diff
@@ -199,39 +404,63 @@ func (r *BtpOperatorReconciler) HandleInitialState(ctx context.Context, cr *v1al
 	return r.UpdateBtpOperatorStatus(ctx, cr, types.StateProcessing, Initialized, "Initialized")
 }
 
+func (r *BtpOperatorReconciler) CheckIfReasonIsSet(cr *v1alpha1.BtpOperator, reason Reason) bool {
+	var condition *v1.Condition
+	if len(cr.Status.Conditions) > 0 {
+		condition = cr.Status.Conditions[0]
+	}
+	return condition != nil && condition.Reason == string(reason)
+}
+
 func (r *BtpOperatorReconciler) HandleProcessingState(ctx context.Context, cr *v1alpha1.BtpOperator) error {
 	logger := log.FromContext(ctx)
-	logger.Info("Handling Processing state")
+	switch {
+	case r.CheckIfReasonIsSet(cr, UpdateCheck):
+		logger.Info("update check processing")
 
-	secret, err := r.getRequiredSecret(ctx)
-	if err != nil {
-		logger.Error(err, "while getting the required Secret")
-		return r.UpdateBtpOperatorStatus(ctx, cr, types.StateError, MissingSecret, "Secret resource not found")
-	}
+		configMap, err := r.getBtpManagerConfigMap(ctx)
+		if err != nil {
+			return r.UpdateBtpOperatorStatus(ctx, cr, types.StateError, GettingConfigMapFailed, err.Error())
+		}
 
-	if err = r.verifySecret(secret); err != nil {
-		logger.Error(err, "while verifying the required Secret")
-		return r.UpdateBtpOperatorStatus(ctx, cr, types.StateError, InvalidSecret, "Secret validation failed")
-	}
+		versionChanged, err := r.storeChartDetails(ctx, configMap)
+		if err != nil {
+			return r.UpdateBtpOperatorStatus(ctx, cr, types.StateError, StoringChartDetailsFailed, err.Error())
+		}
 
-	r.addTempLabelsToCr(cr)
+		if versionChanged {
+			if reason, err := r.handleUpdate(ctx, cr, configMap); err != nil {
+				return r.UpdateBtpOperatorStatus(ctx, cr, types.StateError, reason, err.Error())
+			}
+			r.updateCheckDone = true
+			return r.UpdateBtpOperatorStatus(ctx, cr, types.StateReady, UpdateDone, "Update dome")
+		} else {
+			r.updateCheckDone = true
+			return r.UpdateBtpOperatorStatus(ctx, cr, types.StateReady, UpdateCheckSucceded, "UpdateCheck done without need to update")
+		}
+	default:
+		logger.Info("Handling Processing state")
```

Comment:
if we keep `Handling Processing state` log at the beginning then we could log `provisioning` here